parsers/mcmaster-parser-2.js

```diff
@@ -19,6 +19,53 @@ function convertTo24Hour(time) {
 }
 
 /**
+ * Takes any date and turns it into the correct format, so javascript gets the right date.
+ * MM/DD/YYYY is the format the javascript object reads or YYYY/MM/DD
+ * @param {String} startDate
+ * @param {String} endDate
+ * @return {String Array} [startDate,endDate]  woth format MM/DD/YYYY or YYYY/MM/DD
+ */
+function fixDateFormat(sem_start, sem_end) {
+    if (sem_start.indexOf('-') > -1) {
+        return [sem_start,sem_end]
+    }
+    //assume format is either YYYY/MM/DD TO START OR MM/DD/YYYY
+    startDateSplit = sem_start.split("/");
+    endDateSplit = sem_end.split("/");
+
+    startMonth = parseInt(startDateSplit[0]);
+    startDay = parseInt(startDateSplit[1]);
+    startYear = parseInt(startDateSplit[2]);
+
+    endMonth = parseInt(endDateSplit[0]);
+    endDay = parseInt(endDateSplit[1]);
+    endYear = parseInt(endDateSplit[2]);
+
+    // This means the format is YYYY/MM/DD, because the 0th index is a year
+    if(startDateSplit[0].length === 4){
```

Comment:
'startMonth' is an int so the .length wouldnt work. I'll change to 'startMonth.toString().length'